tests/baselines/reference/complexRecursiveCollections.errors.txt

```diff
@@ -445,7 +440,6 @@ tests/cases/compiler/immutable.ts(391,22): error TS2430: Interface 'Set<T>' inco
 !!! error TS2430:   Types of property 'toSeq' are incompatible.
 !!! error TS2430:     Type '() => Set<T>' is not assignable to type '() => this'.
 !!! error TS2430:       Type 'Set<T>' is not assignable to type 'this'.
-!!! error TS2430:         'Set<T>' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'Set<T>'.

```

Comment:
Marking this for review

Reply:
It's fine - that we issues the elaboration before here was incorrect, apparently. I looked into it - since we were getting the `immediatebaseConstraint` before, it wasn;t instantiated with the proper `this` type, so `Set<T>` was assignable to `Set<T>`, and we chose to elaborate. Now that we're properly using `getBaseConstraintOfType`, the constraint is `Set<T>` with `this = this` (sshhhh, yes I know that's confusing). `Set<T>` is _not_ assignable to that, so we don't elaborate.

Reply:
Thanks @weswigham !